src/config/config.go

```diff
@@ -47,35 +50,58 @@ func printHelpAndExit(err error) {
 
 func ProcessArgs() *Config {
 	ret := &Config{}
-	flag.StringVar(&ret.Role, "role", string(models.HostRoleMaster), "The node role")
-	flag.StringVar(&ret.ClusterID, "cluster-id", "", "The cluster id")
-	flag.StringVar(&ret.InfraEnvID, "infra-env-id", "", "This host infra env id")
-	flag.StringVar(&ret.HostID, "host-id", "", "This host id")
-	flag.StringVar(&ret.Device, "boot-device", "", "The boot device")
-	flag.StringVar(&ret.URL, "url", "", "The BM inventory URL, including a scheme and optionally a port (overrides the host and port arguments")
-	flag.StringVar(&ret.OpenshiftVersion, "openshift-version", "4.4", "Openshift version to install")
-	flag.StringVar(&ret.MCOImage, "mco-image", "", "MCO image to install")
-	flag.BoolVar(&ret.Verbose, "verbose", false, "Increase verbosity, set log level to debug")
-	flag.StringVar(&ret.ControllerImage, "controller-image", "quay.io/ocpmetal/assisted-installer-controller:latest",
+	flagSet := flag.NewFlagSet("flagset", flag.ContinueOnError)
```

Comment:
Why `ContinueOnError` here?
The [previous implementation used `ExitOnError`](https://cs.opensource.google/go/go/+/refs/tags/go1.18.2:src/flag/flag.go;drc=7791e934c882fd103357448aee0fd577b20013ce;l=1147)
Maybe just use that definition as a reference and do `flagSet := flag.NewFlagSet(os.Args[0], flag.ExitOnError)`?

Also, you don't even need to define the flagSet now as we're only doing this handling once, but I personally prefer less global state.